airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -65,6 +65,11 @@ class Cursor(ABC):
     def state(self) -> MutableMapping[str, Any]:
         ...
 
+    @property
+    @abstractmethod
+    def cursor_field(self) -> CursorField:
```

Comment:
Rather than change the `ConcurrentCursor` interface to add a `cursor()` method, could we instead change the constructor above to take in a `cursor_field`. Based on where we'll be instantiating the new CursorPartitionGenerator from the new `ConcurrentDeclarativeSource`, we should be able to pass in the `cursor_field`.

My main thought process here is prioritizing a more tightly scoped `Cursor` interface with only the methods we need. Because if we continue to implement more new cursors it might started to get harder to use. And right now we have an easy way to just pass `field_name` via the constructor even if `__init__` needs to have more parameters

Reply:
For me, this sounds fine. With this, I just want to avoid changes in the cursor class methods.